cdap-app-fabric/src/main/java/io/cdap/cdap/app/store/Store.java

```diff
@@ -362,6 +372,14 @@ boolean scanApplications(ScanApplicationsRequest request, int txBatchSize,
    */
   Collection<ApplicationSpecification> getAllAppVersions(ApplicationId id);
 
+  /**
+   * Returns the latest version of an application
```

Comment:
also include what it does if the application doesn't exist (return null? throw an exception?)